client/src/it/java/org/camunda/bpm/client/task/ExternalTaskCustomExtensionPropertiesIT.java

```diff
@@ -0,0 +1,116 @@
+import org.camunda.bpm.client.util.RecordingExternalTaskHandler;
+import org.camunda.bpm.model.bpmn.Bpmn;
+import org.camunda.bpm.model.bpmn.BpmnModelInstance;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+public class ExternalTaskCustomExtensionPropertiesIT {
+
+  protected BpmnModelInstance externalTaskProcess;
+
+  protected ClientRule clientRule = new ClientRule();
+  protected EngineRule engineRule = new EngineRule();
+
+  @Rule
+  public RuleChain ruleChain = RuleChain.outerRule(engineRule).around(clientRule);
+
+  protected ExternalTaskClient client;
+
+  protected ProcessDefinitionDto processDefinition;
+  protected ProcessInstanceDto processInstance;
+
+  protected RecordingExternalTaskHandler handler = new RecordingExternalTaskHandler();
+
+  @Before
+  public void setup() {
+    client = clientRule.client();
+    handler.clear();
+  }
+
+  @Test
+  public void shouldReceiveCustomExtensionProperties() {
+    // given
+    externalTaskProcess = Bpmn.readModelFromStream(
+        getClass().getClassLoader().getResourceAsStream("model/ExternalTaskCustomExtensionPropertiesIT.oneExternalTaskWithCustomProperties.bpmn20.xml"));
+    processDefinition = engineRule.deploy(externalTaskProcess).get(0);
+    engineRule.startProcessInstance(processDefinition.getId());
+
+    // when
+    client.subscribe(ProcessModels.EXTERNAL_TASK_TOPIC_FOO).includeExtensionProperties(true).handler(handler).open();
+
+    // then
+    clientRule.waitForFetchAndLockUntil(() -> !handler.getHandledTasks().isEmpty());
+    ExternalTask task = handler.getHandledTasks().get(0);
+
+    assertThat(task.getExtensionProperties()).containsOnly(entry("property1", "value1"), entry("property2", "value2"), entry("property3", "value3"));
+    assertThat(task.getExtensionProperty("property1")).isEqualTo("value1");
+    assertThat(task.getExtensionProperty("property2")).isEqualTo("value2");
+    assertThat(task.getExtensionProperty("property3")).isEqualTo("value3");
+    assertThat(task.getExtensionProperty("property4")).isNull();
+  }
+
+  @Test
+  public void shouldReceiveEmptyCustomExtensionProperties() {
+    // given
+    externalTaskProcess = Bpmn.readModelFromStream(
+        getClass().getClassLoader().getResourceAsStream("model/ExternalTaskCustomExtensionPropertiesIT.oneExternalTaskWithoutCustomProperties.bpmn20.xml"));
+    processDefinition = engineRule.deploy(externalTaskProcess).get(0);
+    engineRule.startProcessInstance(processDefinition.getId());
+
+    // when
+    client.subscribe(ProcessModels.EXTERNAL_TASK_TOPIC_FOO).includeExtensionProperties(true).handler(handler).open();
+
+    // then
+    clientRule.waitForFetchAndLockUntil(() -> !handler.getHandledTasks().isEmpty());
+    ExternalTask task = handler.getHandledTasks().get(0);
+
+    assertThat(task.getExtensionProperties()).isEmpty();
+  }
+
+  @Test
+  public void shouldNotReceiveCustomExtensionProperties() {
```

Comment:
That needs to be changed to reflect the "never null" contract for the properties map.
Might be that you need to change something in the implementation as well to fulfill that contract.